src/custom-lists/reducer.js

```diff
@@ -0,0 +1,172 @@
+import { createReducer } from 'redux-act'
+import union from 'lodash/fp/union'
+
+import * as actions from './actions'
+
+const defaultState = {
+    listCount: null,
+    activeListIndex: -1,
+    listFilterIndex: null,
+    /** list will contain some data like
+     * {
+     *  id: Unique ID of the list
+     *  listName: @[string],
+     *  pages: [Array of urls].
+     * }
+     *
+     */
+    lists: [],
+    deleteConfirmProps: {
+        isShown: false,
+        id: undefined,
+        // Used to keep track of any particular result (use index)
+        deleting: undefined,
+    },
+    urlDragged: '',
+    showCreateListForm: false,
+    showCommonNameWarning: false,
+}
+
+const fetchAllLists = (state, lists) => ({
+    ...state,
+    lists,
+    tempLists: lists,
+})
+
+const createList = (state, list) => ({
+    ...state,
+    lists: [...state.lists, list],
+})
+
+const updateList = (state, { value, index }) => ({
+    ...state,
+    lists: [
+        ...state.lists.slice(0, index),
+        {
+            ...state.lists[index],
+            name: value,
+        },
+        ...state.lists.slice(index + 1),
+    ],
+})
+
+const deleteList = (state, { id, index }) => ({
+    ...state,
+    lists: [...state.lists.slice(0, index), ...state.lists.slice(index + 1)],
+})
+
+const addPageToList = (state, { url, index }) => {
+    // TODO: Maybe use id instead
+    const { lists } = state
+    const list = lists[index]
+
+    if (list.pages.indexOf(url) > -1) return state
```

Comment:
`Array.prototype.includes` would be more appropriate here than `Array.prototype.indexOf`. 

On the next line, you're doing the condition `typeof filters === 'string'`, but I cannot see the `filters` name being assigned anywhere. Should this be `url`? I would also encourage to try and think if it would ever be `string[]` type or it's always going to be `string` type? The less dynamically typed values needed to be dealt with, the better (although there could be a good reason you're doing it like that, so use your best judgement)